File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [1] * len(nums)
        pre = 1
        for i in range(len(nums)):
            res[i] = pre
            pre *= nums[i]
            # nums = [1,2,4,6]
        count = 1
        for i in range(len(nums) - 1, -1, -1):
            # res = [1,1,2,8] 8
            res[i] *= count
            count *= nums[i]

        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        /*
        Input: nums = [2,20,4,10,3,4,5]
        figure out if the item is a head or not
        */
        int res = 0;
        unordered_set<int> nums_set(nums.begin(), nums.end());

        for (auto n: nums_set)
        {
            if (nums_set.find(n-1) == nums_set.end())
            {
                int length = 1;
                while (nums_set.find(n+1) != nums_set.end())
                {
                    length++;
                    n++;
                }
                res = max(length, res);
            }
        } 
        return res;

    }
};
